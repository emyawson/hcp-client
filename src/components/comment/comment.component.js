import React from 'react';

import { SpeechBubbleIcon } from 'src/assets/icons';
import { colors } from 'src/core/styles';
import { RenderIf } from 'src/utils';

import {
  CommentCardDiv,
  CommentWrapperDiv,
  CommentContentDiv,
  CommentIconDiv,
  CommentIconBadge,
  CommentHeaderDiv,
  CommentBodyP,
  CommentHeadline,
  CommentDateSpan,
} from './comment.style';

export const Comment = ({
  date,
  isExpanded = false,
  message,
  showIcon = true,
  title,
}) => (
  <CommentCardDiv>
    <CommentWrapperDiv isExpanded={isExpanded}>
      <RenderIf validate={showIcon}>
        <CommentIconDiv isExpanded={isExpanded}>
          <CommentIconBadge
            bgColor={colors.trafficRed}
            icon={<SpeechBubbleIcon height={20} fillColor={colors.white} />}
            size={38}
          />
        </CommentIconDiv>
      </RenderIf>
      <CommentContentDiv>
        <CommentHeaderDiv>
          <CommentHeadline isExpanded={isExpanded}>{title}</CommentHeadline>
          <CommentDateSpan>- {date}</CommentDateSpan>
        </CommentHeaderDiv>
        <CommentBodyP>{message}</CommentBodyP>
      </CommentContentDiv>
    </CommentWrapperDiv>
  </CommentCardDiv>
);
