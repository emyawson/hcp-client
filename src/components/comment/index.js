export * from './comment.component.js';
