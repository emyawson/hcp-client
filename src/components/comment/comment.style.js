import styled from 'styled-components';

import { Badge } from 'src/components';
import { borderRadius } from 'src/core/styles';
import { convertPxToRem } from 'src/utils';

export const CommentCardDiv = styled.div`
  background-color: ${props => props.theme.colors.lavender};
  border: 1px solid ${props => props.theme.colors.quartzBlue};
  border-radius: ${borderRadius.three};
  color: ${props => props.theme.colors.charcoal};
  overflow: hidden;
`;

const borderSize = convertPxToRem(3);

export const CommentWrapperDiv = styled.div`
  border-left: ${props =>
    props.isExpanded
      ? `${borderSize} solid ${props.theme.colors.trafficRed}`
      : 'none'};
  border-top: ${props =>
    !props.isExpanded
      ? `${borderSize} solid ${props.theme.colors.trafficRed}`
      : 'none'};
  display: flex;
  padding: ${props => props.theme.spacing.three}
    ${props =>
      props.isExpanded ? props.theme.spacing.four : props.theme.spacing.three};
`;

export const CommentContentDiv = styled.div`
  width: 100%;
`;

export const CommentIconDiv = styled.div`
  align-items: center;
  border-right: 1px solid ${props => props.theme.colors.quartzBlue};
  display: flex;
  margin-right: ${props =>
    props.isExpanded ? props.theme.spacing.four : props.theme.spacing.three};
  padding-right: ${props =>
    props.isExpanded ? props.theme.spacing.four : props.theme.spacing.three};
`;

export const CommentIconBadge = styled(Badge)`
  padding: ${borderSize};
`;

export const CommentHeaderDiv = styled.div`
  align-items: baseline;
  display: flex;
  margin: 0 auto ${props => props.theme.spacing.two};
  width: 100%;
`;

const CommentHeaderText = styled.span`
  color: ${props => props.theme.colors.blueMarine5};
  font-weight: ${props => props.theme.fontWeights.semiBold};
  margin: 0;
`;

export const CommentHeadline = CommentHeaderText.withComponent('h2').extend`
  font-size: ${props =>
    props.isExpanded
      ? props.theme.fontSize.subheading
      : props.theme.fontSize.p};
  line-height: 1;
  margin-right: ${props => props.theme.spacing.one};
`;

export const CommentDateSpan = CommentHeaderText.extend`
  font-size: ${props => props.theme.fontSize.label};
  text-transform: uppercase;
`;

export const CommentBodyP = styled.p`
  font-size: ${props => props.theme.fontSize.p};
  line-height: 1.75;
  margin: 0;
`;
