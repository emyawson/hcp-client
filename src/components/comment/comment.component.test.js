import React from 'react';
import { shallow } from 'enzyme';

import { Comment } from './comment.component';

describe('Comment Component tests', () => {
  const mockProps = {
    date: 'June 01 2018',
    isExpanded: false,
    title: 'Force Status',
    message:
      'The patient has not achieved the required conditions and must see the practitioner for the strips.',
    showIcon: false,
  };
  it('should render correctly', () => {
    const wrapper = shallow(<Comment {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
