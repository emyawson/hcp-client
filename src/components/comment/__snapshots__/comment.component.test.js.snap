// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component tests should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    date="June 01 2018"
    isExpanded={false}
    message="The patient has not achieved the required conditions and must see the practitioner for the strips."
    showIcon={false}
    title="Force Status"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div
        isExpanded={false}
      >
        <f1
          validate={false}
        >
          <styled.div
            isExpanded={false}
          >
            <Styled(Badge)
              bgColor="#CF021B"
              icon={
                <Unknown
                  fillColor="#FFF"
                  height={20}
                />
              }
              size={38}
            />
          </styled.div>
        </f1>
        <styled.div>
          <styled.div>
            <styled.h2
              isExpanded={false}
            >
              Force Status
            </styled.h2>
            <styled.span>
              - 
              June 01 2018
            </styled.span>
          </styled.div>
          <styled.p>
            The patient has not achieved the required conditions and must see the practitioner for the strips.
          </styled.p>
        </styled.div>
      </styled.div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <f1
            validate={false}
          >
            <styled.div
              isExpanded={false}
            >
              <Styled(Badge)
                bgColor="#CF021B"
                icon={
                  <Unknown
                    fillColor="#FFF"
                    height={20}
                  />
                }
                size={38}
              />
            </styled.div>
          </f1>,
          <styled.div>
            <styled.div>
              <styled.h2
                isExpanded={false}
              >
                Force Status
              </styled.h2>
              <styled.span>
                - 
                June 01 2018
              </styled.span>
            </styled.div>
            <styled.p>
              The patient has not achieved the required conditions and must see the practitioner for the strips.
            </styled.p>
          </styled.div>,
        ],
        "isExpanded": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <styled.div
              isExpanded={false}
            >
              <Styled(Badge)
                bgColor="#CF021B"
                icon={
                  <Unknown
                    fillColor="#FFF"
                    height={20}
                  />
                }
                size={38}
              />
            </styled.div>,
            "validate": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Badge)
                bgColor="#CF021B"
                icon={
                  <Unknown
                    fillColor="#FFF"
                    height={20}
                  />
                }
                size={38}
              />,
              "isExpanded": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "bgColor": "#CF021B",
                "icon": <Unknown
                  fillColor="#FFF"
                  height={20}
                />,
                "size": 38,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.h2
                  isExpanded={false}
                >
                  Force Status
                </styled.h2>
                <styled.span>
                  - 
                  June 01 2018
                </styled.span>
              </styled.div>,
              <styled.p>
                The patient has not achieved the required conditions and must see the practitioner for the strips.
              </styled.p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.h2
                    isExpanded={false}
                  >
                    Force Status
                  </styled.h2>,
                  <styled.span>
                    - 
                    June 01 2018
                  </styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Force Status",
                    "isExpanded": false,
                  },
                  "ref": null,
                  "rendered": "Force Status",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "- ",
                      "June 01 2018",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "- ",
                    "June 01 2018",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": "The patient has not achieved the required conditions and must see the practitioner for the strips.",
              },
              "ref": null,
              "rendered": "The patient has not achieved the required conditions and must see the practitioner for the strips.",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div
          isExpanded={false}
        >
          <f1
            validate={false}
          >
            <styled.div
              isExpanded={false}
            >
              <Styled(Badge)
                bgColor="#CF021B"
                icon={
                  <Unknown
                    fillColor="#FFF"
                    height={20}
                  />
                }
                size={38}
              />
            </styled.div>
          </f1>
          <styled.div>
            <styled.div>
              <styled.h2
                isExpanded={false}
              >
                Force Status
              </styled.h2>
              <styled.span>
                - 
                June 01 2018
              </styled.span>
            </styled.div>
            <styled.p>
              The patient has not achieved the required conditions and must see the practitioner for the strips.
            </styled.p>
          </styled.div>
        </styled.div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <f1
              validate={false}
            >
              <styled.div
                isExpanded={false}
              >
                <Styled(Badge)
                  bgColor="#CF021B"
                  icon={
                    <Unknown
                      fillColor="#FFF"
                      height={20}
                    />
                  }
                  size={38}
                />
              </styled.div>
            </f1>,
            <styled.div>
              <styled.div>
                <styled.h2
                  isExpanded={false}
                >
                  Force Status
                </styled.h2>
                <styled.span>
                  - 
                  June 01 2018
                </styled.span>
              </styled.div>
              <styled.p>
                The patient has not achieved the required conditions and must see the practitioner for the strips.
              </styled.p>
            </styled.div>,
          ],
          "isExpanded": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "children": <styled.div
                isExpanded={false}
              >
                <Styled(Badge)
                  bgColor="#CF021B"
                  icon={
                    <Unknown
                      fillColor="#FFF"
                      height={20}
                    />
                  }
                  size={38}
                />
              </styled.div>,
              "validate": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Badge)
                  bgColor="#CF021B"
                  icon={
                    <Unknown
                      fillColor="#FFF"
                      height={20}
                    />
                  }
                  size={38}
                />,
                "isExpanded": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "bgColor": "#CF021B",
                  "icon": <Unknown
                    fillColor="#FFF"
                    height={20}
                  />,
                  "size": 38,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.div>
                  <styled.h2
                    isExpanded={false}
                  >
                    Force Status
                  </styled.h2>
                  <styled.span>
                    - 
                    June 01 2018
                  </styled.span>
                </styled.div>,
                <styled.p>
                  The patient has not achieved the required conditions and must see the practitioner for the strips.
                </styled.p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.h2
                      isExpanded={false}
                    >
                      Force Status
                    </styled.h2>,
                    <styled.span>
                      - 
                      June 01 2018
                    </styled.span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Force Status",
                      "isExpanded": false,
                    },
                    "ref": null,
                    "rendered": "Force Status",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": Array [
                        "- ",
                        "June 01 2018",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "- ",
                      "June 01 2018",
                    ],
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "The patient has not achieved the required conditions and must see the practitioner for the strips.",
                },
                "ref": null,
                "rendered": "The patient has not achieved the required conditions and must see the practitioner for the strips.",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
